src/main/scala/com/evolutiongaming/util/Validation.scala

```diff
@@ -184,6 +185,28 @@ object Validation {
   implicit class IterableOps[A](val self: Iterable[A]) extends AnyVal {
 
     def ?>>[B](left: => B): Either[B, Iterable[A]] = if (self.isEmpty) Left(left) else Right(self)
+
+    /**
+      * Checks that all elements in `self` are valid by applying the `test` function to each.
+      * If so, returns a [[scala.util.Right]] otherwise returns the first [[scala.util.Left]]
+      * encountered.
+      *
+      * @param test function to apply to each element
+      * @tparam T type of error the `test` function returns
+      */
+    def allValid[T](test: A => Either[T, Unit]): Either[T, Unit] = {
```

Comment:
Or do you mean we could have something more useful in the `Right` part other than the `Unit`?
E.g. something like `def allValid[L, R]`(test: A: Either[L, R]): Either[L, Iterable[R]]`